src/argilla/server/contexts/datasets.py

```diff
@@ -905,6 +920,7 @@ async def update_record(
 
         if needs_search_engine_update:
             await record.dataset.awaitable_attrs.vectors_settings
+            await refresh_records([record])
             await search_engine.index_records(record.dataset, [record])
```

Comment:
As said before:

```suggestion
            await search_engine.index_records(
                record.dataset,
                [await _preload_record_associations(record)]
            )
```

Or:

```suggestion
            await _preload_record_associations(record)
            await search_engine.index_records(record.dataset, [record])
```

Reply:
For the first approach, that new function should return the loaded record list. We can review it together